value-processor/src/org/immutables/value/processor/Immutables.generator

```diff
@@ -4239,7 +4239,7 @@ private static <K, V> java.util.Map<K, V> createUnmodifiableMap(boolean checkNul
     V v = e.getValue();
     if (checkNulls) {
       [requireNonNull type](k, "key");
-      [requireNonNull type](v, "value");
+      [requireNonNull type](v, "value for key: " + k);
```

Comment:
I wonder if string concatenation can be costly here ?
Perhaps use a version with the [Supplier](https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-java.util.function.Supplier-) ?

Reply:
I tried using `[requireNonNull type](v, () -> "value for key: " + k);` already but that change is slightly more involved as it probably requires to change a bunch of other places. So maybe this can be done in a follow-up?

Reply:
I was thinking the same, I cannot predict exact impact, but most of the precondition checks are highly optimized for almost zero cost noop case, this up-front concatenation have high chance to skew inlining etc. Maybe we can just prefix it with, `if (v == null) `, like

```java
if (v == null) [requireNonNull type](v, "value for key: " + k)
```

Reply:
we will pay for branching in this case, guess it's an acceptable tradeoff

Reply:
fixed